### PR TITLE
ETH chain simulator in tests

### PR DESCRIPTION
**Pull request by sstanculeanu**
DO NOT MERGE yet, until proper contracts update

### CODE REVIEW

File: integrationTests/relayers/ethToMultiversXWithChainSimulator_test.go [hunk 1]
```diff
@@ -193,15 +206,26 @@ func TestRelayersShouldExecuteTransfersFromEthToMultiversXWithChainSimulator(t *
 		pk: ownerPK,
 		sk: ownerSK,
 	}
+	erc20ContractsHolder := createMockErc20ContractsHolder([]common.Address{ethGenericTokenAddress}, ethSafeAddress, []*big.Int{mintAmount})
```
Comment (dragos-rebegea) [inline integrationTests/relayers/ethToMultiversXWithChainSimulator_test.go:209]: we can use a real component of Erc20SafeContractsHolder

Comment (sstanculeanu) [reply integrationTests/relayers/ethToMultiversXWithChainSimulator_test.go:209]: pushed

File: integrationTests/relayers/ethToMultiversXWithChainSimulator_test.go [hunk 2]
```diff
@@ -930,3 +963,154 @@ func submitAggregatorBatch(t *testing.T, ctx context.Context, multiversXProxyWit
...
+) (*backends.SimulatedBackend, blockchainClient, *contract.ERC20Safe, common.Address, *contract.Bridge, common.Address, *contract.GenericERC20, common.Address) {
+	addr := map[common.Address]ethCore.GenesisAccount{
+		ethOwnerAddr:     {Balance: new(big.Int).Lsh(big.NewInt(1), 100)},
+		ethDepositorAddr: {Balance: new(big.Int).Lsh(big.NewInt(1), 100)},
+	}
+	for _, relayerKeys := range relayersKeys {
+		addr[relayerKeys.ethAddress] = ethCore.GenesisAccount{Balance: new(big.Int).Lsh(big.NewInt(1), 100)}
+	}
+	alloc := ethCore.GenesisAlloc(addr)
+	simulatedETHChain := backends.NewSimulatedBackend(alloc, 9000000)
```
Comment (iulianpascalau) [inline integrationTests/relayers/ethToMultiversXWithChainSimulator_test.go:982]: can we have a const/var for `9000000` ?

File: integrationTests/relayers/testdata/contracts/eth/bridgeBytecode.txt [hunk 3]
```diff
@@ -0,0 +1 @@
...57611e8f611e67565b5060010190565b80516001600160701b0381168114611ead57600080fd5b919050565b805167ffffffffffffffff81168114611ead57600080fd5b600060808284031215611edc57600080fd5b6040516080810181811067ffffffffffffffff82111715611eff57611eff611b9a565b604052611f0b83611e96565b8152611f1960208401611eb2565b6020820152611f2a60408401611eb2565b6040820152606083015161ffff81168114611f4457600080fd5b60608201529392505050565b805160058110611ead57600080fd5b60006020808385031215611f7257600080fd5b825167ffffffffffffffff811115611f8957600080fd5b8301601f81018513611f9a57600080fd5b8051611fa8611c3c82611c0a565b81815260c09182028301840191848201919088841115611fc757600080fd5b938501935b8385101561204f5780858a031215611fe45760008081fd5b611fec611bb0565b611ff586611e96565b8152868601516120048161188e565b81880152604086810151908201526060808701516120218161188e565b908201526080868101519082015260a061203c818801611f50565b9082015283529384019391850191611fcc565b50979650505050505050565b8082018082111561091457610914611e67565b60008351612080818460208801611d79565b9190910191825250602001919050565b8181038181111561091457610914611e67565b634e487b7160e01b600052603160045260246000fd5b60ff818116838216019081111561091457610914611e6756fea164736f6c6343000814000a
```
Comment (iulianpascalau) [inline integrationTests/relayers/testdata/contracts/eth/bridgeBytecode.txt:1]: this file contains hexed byte code. Should we rename it to bridgeBytecode.hex?
Valid for all .txt files

**Review by iulianpascalau — COMMENTED** [review]
GJ :+1:

File: integrationTests/relayers/testdata/contracts/eth/bridgeBytecode.txt (hunk 3 above)
Comment (sstanculeanu) [reply integrationTests/relayers/testdata/contracts/eth/bridgeBytecode.txt:1]: updated

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**